github/sub_issue_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2013 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestSubIssuesService_Add(t *testing.T) {
+	t.Parallel()
+	client, mux, _ := setup(t)
+
+	input := &SubIssueRequest{SubIssueID: 42}
+
+	mux.HandleFunc("/repos/o/r/issues/1/sub_issues", func(w http.ResponseWriter, r *http.Request) {
+		v := new(SubIssueRequest)
+		assertNilError(t, json.NewDecoder(r.Body).Decode(v))
+
+		testMethod(t, r, "POST")
+		if !cmp.Equal(v, input) {
+			t.Errorf("Request body = %+v, want %+v", v, input)
+		}
+
+		fmt.Fprint(w, `{"id":42, "number":1}`)
+	})
+
+	ctx := context.Background()
+	got, _, err := client.SubIssue.Add(ctx, "o", "r", 1, input)
+	if err != nil {
+		t.Errorf("SubIssues.Add returned error: %v", err)
+	}
+
+	want := &SubIssue{Number: Ptr(1), ID: Int64(42)}
+	if !cmp.Equal(got, want) {
+		t.Errorf("SubIssues.Add = %+v, want %+v", got, want)
+	}
+}
```

Comment:
For all these tests, please add the appropriate test helpers that you will find throughout this repo... e.g. `testBadOptions` and `testNewRequestAndDoFailure`